src/main/java/org/idea/plugin/atg/usage/implicit/ActorImplicitUsageProvider.java

```diff
@@ -0,0 +1,176 @@
+import com.intellij.psi.xml.XmlAttribute;
+import com.intellij.psi.xml.XmlTag;
+import com.intellij.util.indexing.FileBasedIndex;
+import org.apache.commons.lang.StringUtils;
+import org.idea.plugin.atg.filter.ExcludedDirectoryVirtualFileFilter;
+import org.idea.plugin.atg.index.XmlActorIndexExtension;
+import org.idea.plugin.atg.iterator.PropertiesContentIterator;
+import org.idea.plugin.atg.visitor.XmlPsiRecursiveElementVisitor;
+
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+
+public class ActorImplicitUsageProvider implements ImplicitUsageProvider {
+
+    private static final String PROPERTY_CLASS = "$class";
+    private static final String XML_TAG_COMPONENT = "component";
+    private static final String XML_TAG_FORM = "form";
+    private static final String DOT_DELIMITER = ".";
+    private static final String XML_TAG_INPUT = "input";
+    private static final String XML_ATTRIBUTE_CLASS_NAME = "class-name";
+    private static final String XML_ATTRIBUTE_NAME = "name";
+    private static final String EMPTY_STRING = "";
+    private static final String PREFIX_HANDLE = "handle";
+
+    @Override
+    public boolean isImplicitRead(PsiElement element) {
+        return false;
+    }
+
+    @Override
+    public boolean isImplicitWrite(PsiElement element) {
+        return false;
+    }
+
+    @Override
+    public boolean isImplicitUsage(PsiElement element) {
+        boolean result = false;
+        if (element instanceof PsiMethod) {
+            PsiMethod inspectMethod = (PsiMethod) element;
+            Project project = inspectMethod.getProject();
+            Collection<String> xmlFileName = FileBasedIndex.getInstance().getAllKeys(XmlActorIndexExtension.NAME, project);
+            List<String> pathToFiles = xmlFileName.stream()
+                    .flatMap(nameOfFile -> FileBasedIndex.getInstance()
+                            .getValues(XmlActorIndexExtension.NAME, nameOfFile, GlobalSearchScope.allScope(project)).stream())
+                    .collect(Collectors.toList());
+            List<String> componentPaths;
+            ProjectFileIndex projectFileIndex = ProjectRootManager.getInstance(project).getFileIndex();
+            if (((PsiMethod) element).getName().startsWith(PREFIX_HANDLE)) {
+                componentPaths = findComponentNameWithXmlTag(pathToFiles, XML_TAG_FORM, project, inspectMethod);
+            } else {
+                componentPaths = findComponentNameWithXmlTag(pathToFiles, XML_TAG_COMPONENT, project, inspectMethod);
+            }
+
+            // TODO refactor get property files from index
+            result = componentPaths.stream().anyMatch(componentPath -> {
+                List<VirtualFile> propertiesVirtualFiles = new ArrayList<>();
+                PropertiesContentIterator propertiesContentIterator =
+                        new PropertiesContentIterator(componentPath, propertiesVirtualFiles);
+                projectFileIndex.iterateContent(propertiesContentIterator);
```

Comment:
In my plugin i have example of  usage index for iterating over Xml contents.
Consider to check org.idea.plugin.atg.index.AtgIndexService#getXmlsByName
additional iteration over another index is redundant